addons/sale_timesheet/models/sale_order.py

```diff
@@ -21,24 +21,17 @@ class SaleOrder(models.Model):
     timesheet_total_duration = fields.Integer("Timesheet Total Duration", compute='_compute_timesheet_total_duration', help="Total recorded duration, expressed in the encoding UoM, and rounded to the unit")
 
     def _compute_timesheet_ids(self):
-        timesheet_groups = self.env['account.analytic.line'].sudo().read_group(
-            [('so_line', 'in', self.mapped('order_line').ids), ('project_id', '!=', False)],
-            ['so_line', 'ids:array_agg(id)'],
-            ['so_line'])
-        timesheets_per_sol = {group['so_line'][0]: (group['ids'], group['so_line_count']) for group in timesheet_groups}
+        Timesheet = self.env['account.analytic.line']
+        timesheet_groups = Timesheet.sudo()._read_group(
+            [('order_id', 'in', self.ids), ('project_id', '!=', False)],
+            ['order_id', 'ids:array_agg(id)'],
+            ['order_id'])
+        timesheets_per_sol = {group['order_id'][0]: (group['ids'], group['order_id_count']) for group in timesheet_groups}
```

Comment:
```suggestion
        timesheets_per_so = {
            group['order_id'][0]: (group['ids'], group['order_id_count'])
            for group in self.env['account.analytic.line'].sudo()._read_group(
                [('order_id', 'in', self.ids), ('project_id', '!=', False)],
                ['order_id', 'ids:array_agg(id)'],
                ['order_id']
            )
        }
```
1) Doing it in one step seems clear enough to me.
2) timesheets_per_so*